src/raftstore/store/peer.rs

```diff
@@ -1349,20 +1349,28 @@ impl Peer {
                 // have no effect.
                 self.proposals.clear();
             }
+            if lease_to_be_updated {
+                let first_propose_time = committed_entries
+                    .iter()
+                    .find_map(|entry| self.find_propose_time(entry.get_index(), entry.get_term()));
+                let last_propose_time = committed_entries
+                    .iter()
+                    .rev()
+                    .find_map(|entry| self.find_propose_time(entry.get_index(), entry.get_term()))
+                    .or(first_propose_time);
```

Comment:
`or` is unnecessary. It's the same as `or(None)`.

Reply:
if there is only one committed entry, the first round of `find_propose_time` will consume inner proposal queues.